lib/interactive_viewer.ml

```diff
@@ -127,32 +127,88 @@ let rec split_and_align_hunk hunks mine_acc their_acc =
   | `Their s :: t ->
       split_and_align_hunk t (`Common "" :: mine_acc) (`Their s :: their_acc)
 
-let lines_to_ui lines attr_change =
-  List.map
-    (fun line ->
-      let content, attr =
-        match line with
-        | `Common s -> (s, Notty.A.empty)
-        | `Mine s -> (s, attr_change)
-        | `Their s -> (s, attr_change)
-      in
-      W.string ~attr content)
+let lines_with_numbers lines attr_change line_num prefix =
+  List.mapi
+    (fun index line ->
+      let is_last = index + 1 = List.length lines in
+      match line with
+      | `Common s when s = "" && is_last -> ("", Notty.A.empty)
+      | `Mine s when s = "" && is_last -> ("", Notty.A.empty)
+      | `Their s when s = "" && is_last -> ("", Notty.A.empty)
```

Comment:
Why all those special casing here? When I read a diff, I'd like to know when there is an empty last line!